querybook/webapp/redux/board/action.ts

```diff
@@ -328,3 +327,207 @@ export function setCurrentBoardId(boardId: number): ThunkResult<Promise<void>> {
         });
     };
 }
+
+export function getBoardEditors(
+    boardId: number
+): ThunkResult<Promise<IBoardEditor[]>> {
+    return async (dispatch) => {
+        const { data } = await BoardEditorResource.get(boardId);
+
+        dispatch({
+            type: '@@board/RECEIVE_BOARD_EDITORS',
+            payload: {
+                boardId,
+                editors: data,
+            },
+        });
+
+        return data;
+    };
+}
+
+export function addBoardEditors(
+    boardId: number,
+    uid: number,
+    permission: Permission
+): ThunkResult<Promise<IBoardEditor>> {
+    return async (dispatch, getState) => {
+        const request = (getState().board.accessRequestsByBoardIdUserId[
+            boardId
+        ] || {})[uid];
+        const { read, write } = permissionToReadWrite(permission);
+        const { data } = await BoardEditorResource.create(
+            boardId,
+            uid,
+            read,
+            write
+        );
+        if (request) {
+            dispatch({
+                type: '@@board/REMOVE_BOARD_ACCESS_REQUEST',
+                payload: {
+                    boardId,
+                    uid,
+                },
+            });
+        }
+        dispatch({
+            type: '@@board/RECEIVE_BOARD_EDITOR',
+            payload: {
+                boardId,
+                editor: data,
+            },
+        });
+
+        return data;
+    };
+}
+
+export function updateBoardEditors(
+    boardId: number,
+    uid: number,
+    read: boolean,
+    write: boolean
+): ThunkResult<Promise<IBoardEditor>> {
+    return async (dispatch, getState) => {
+        const editor = (getState().board.editorsByBoardIdUserId[boardId] || {})[
+            uid
+        ];
+        if (editor) {
+            const {
+                data,
+            }: {
+                data: IBoardEditor;
```

Comment:
the typecast is unnecessary, it is already provided by `BoardEditorResource.update`